src/test/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporterTest.java

```diff
@@ -0,0 +1,643 @@
+                                                .build())
+                                        .build())
+                                .bids(List.of(
+                                        GreenbidsBids.builder()
+                                                .bidder("seat1")
+                                                .isTimeout(false)
+                                                .hasBid(true)
+                                                .build(),
+                                        GreenbidsBids.builder()
+                                                .bidder("seat2")
+                                                .isTimeout(false)
+                                                .hasBid(false)
+                                                .build()
+                                ))
+                                .build()
+                        ))
+                .auctionElapsed(0L)
+                .build();
+    }
+
+    private static CommonMessage givenCommonMessageForVideo() {
+        return CommonMessage.builder()
+                .version("2.2.0")
+                .auctionId("request1")
+                .sampling(1.0)
+                .pbuid("leparisien")
+                .adUnits(
+                        List.of(GreenbidsAdUnit.builder()
+                                .code("adunitcodevalue")
+                                .unifiedCode(GreenbidsUnifiedCode.builder()
+                                        .value("adunitcodevalue")
+                                        .source("adUnitCodeSource")
+                                        .build())
+                                .mediaTypes(MediaTypes.builder()
+                                        .video(Video.builder()
+                                                .plcmt(1)
+                                                .pos(1)
+                                                .build())
+                                        .build())
+                                .bids(List.of(
+                                        GreenbidsBids.builder()
+                                                .bidder("seat1")
+                                                .isTimeout(false)
+                                                .hasBid(true)
+                                                .build(),
+                                        GreenbidsBids.builder()
+                                                .bidder("seat2")
+                                                .isTimeout(false)
+                                                .hasBid(false)
+                                                .build()
+                                ))
+                                .build()
+                        ))
+                .auctionElapsed(0L)
+                .build();
+    }
+
+    private static CommonMessage givenCommonMessageBannerWithoutFormat() {
```

Comment:
You can use UnaryOperator customizer(examples can be found inside project)

Than `givenCommonMessageForVideo` and `givenCommonMessageForBanner ` and `givenCommonMessageBannerWithoutFormat ` can be simplified

Reply:
Have added customizer for greenbidsBids with

```java
UnaryOperator<GreenbidsBids.GreenbidsBidsBuilder>... greenbidsBidsCustomizers)
```

Also moved `givenExtBanner(Integer w, Integer h, Integer pos)` to a separate method